src/java_tools/buildjar/java/com/google/devtools/build/buildjar/VanillaJavaBuilder.java

```diff
@@ -223,6 +222,15 @@ public VanillaJavaBuilderResult run(List<String> args) throws IOException {
     return new VanillaJavaBuilderResult(ok, output.toString());
   }
 
+  private ImmutableList<Deps.Dependency> collectDependencies(StandardJavaFileManager fileManager) {
+    ImmutableList.Builder<Deps.Dependency> builder = ImmutableList.builder();
+    for (File file : fileManager.getLocation(StandardLocation.CLASS_PATH)) {
+      String path = file.toString();
+      builder.add(Deps.Dependency.newBuilder().setKind(Deps.Dependency.Kind.EXPLICIT).setPath(path).build());
```

Comment:
This is an over-approximation of the deps that were needed, and defeats the point of the optimization IntelliJ is doing. It might make more sense to just provide a way to disable the Intellij plugins' jdeps handling and use the full classpath.

A better solution for `VanillaJavaBuilder` might be to have it use the bytecode-base 
strict deps / jdeps tool: https://github.com/bazelbuild/bazel/blob/423fe65d8a4ea5fdb57fab2f96e1cbc83fb8fed2/src/java_tools/import_deps_checker/java/com/google/devtools/build/importdeps/Main.java#L48